### PR TITLE
Docs: Add linting to rule doc example code blocks (1 of 3) (refs #2271)

### LINKED ISSUE #2271: Lint documentation examples
Split from [#2257 (comment)](https://github.com/eslint/eslint/pull/2257#discussion_r28074602).

With file loaders, we can now lint the example code in documentation files. At the very least, this should catch syntax errors. I think it would be cool if we used configuration comments and actually applied the rule to the valid examples. (Not sure how we would only check the invalid ones for syntax; that'll take some thought.)

<bountysource-plugin>
---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/10795268-lint-documentation-examples?utm_campaign=plugin&utm_content=tracker%2F282608&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F282608&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by IanVS**
This is a work in progress, and will take some time to complete.  However, if anyone has feedback on this, please speak up now before I go through all 170-something rules. :)

One specific item: some of the lines get pretty long with these expected errors.  I'm not sure how that will look on the eslint.org site.  Is there a way to check that?

### CODE REVIEW

**Comment by ilyavolodin** [comment]
Nice! I like it.

**Comment by BYK** [comment]
:astonished: :+1: :tada:

**Comment by IanVS** [comment]
I had to turn off linting for rule `linebreak-style`, which I did by removing the `js` info string from the code fences.  This is because the `\r\n` and `\n` are being treated as code, and not linebreaks.

**Comment by btmills** [comment]
Would moving the line endings into informational comments work? It wouldn't change the meaning of the examples, but it would make them valid JS.

**Comment by IanVS** [comment]
@btmills, I like that idea.  That's what I'll do.

**Comment by nzakas** [comment]
Just a heads up - the longer this sits, the harder it will be to merge. :)

**Comment by IanVS** [comment]
I understand, but I'm not just sitting on it.  I've been spending several hours a night on it, and am maybe 30% done at this point.  :disappointed: 

I've been rebasing periodically, and so far it doesn't have any problems.  I'll give it a good once-over when I'm complete to make sure I've not missed anything that's been added.  I'm hoping to have it done within another week or two.

**Comment by IanVS** [comment]
A bit of a snag: some rule examples require `ecmaFeatures: { modules: true }` and others require `modules: false`.  Until a better solution can be found (maybe involving globbing), I'm removing the info string from the rules which require modules to be turned off.  These are:

* `no-delete-var`
* `no-dupe-args`

**Comment by IanVS** [comment]
I followed @btmills advice for `linebreak-style`, and also added the actual CRLF line endings as appropriate, so that linting works correctly.  To do so, I also needed to add a `.gitattributes` to prevent automatic conversion on that file.

**Comment by nzakas** [comment]
No worries, just wanted to flag that. You might want to consider ways of doing partial commits to limit the merge conflict likelihood. "Big bang" commits are almost always a bad idea. :)

**Comment by IanVS** [comment]
OK.  I'm happy to split into a few commits if that's an option.  It would actually be good to at least see how this looks in some rules on the site, too.  I'm about half-way through, so I'll squash and rebase with just the rule changes so as not to break the build.  After that's merged, I'll finish up the rule changes and finally add the changes to `makefile.js`

**Comment by ilyavolodin** [comment]
@IanVS You can take a look at #397 that's where I migrated all unittests, so had to go through the same trouble as you are going through right now.

**Comment by IanVS** [comment]
OK, tracking in: https://github.com/eslint/eslint/issues/2271#issuecomment-135231393

File: .editorconfig [hunk 1]
```diff
@@ -6,3 +6,10 @@ indent_size = 4
 trim_trailing_whitespace = true
 end_of_line = lf
 insert_final_newline = true
+
```
Comment (ilyavolodin) [inline .editorconfig]: Nice touch!

**Comment by ilyavolodin** [comment]
LGTM for the first batch

**Pull request merged**